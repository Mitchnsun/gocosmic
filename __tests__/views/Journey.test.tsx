import { vi } from 'vitest';

import JourneyContent from '@/views/Journey';

import { render } from '../test-utils';

// Mock @react-three/fiber
vi.mock('@react-three/fiber', () => ({
  Canvas: function MockedCanvas({ children }: { children: React.ReactNode }) {
    return <div data-testid="three-canvas">{children}</div>;
  },
}));

// Mock MovingStarfield component
vi.mock('@/components/Journey/MovingStarfield', () => {
  return {
    default: function MockedMovingStarfield() {
      return <div data-testid="moving-starfield" />;
    },
  };
});

describe('JourneyContent Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should render the journey content with cosmic theme', () => {
    const { container } = render(<JourneyContent />);

    expect(container).toMatchSnapshot();
  });
});
