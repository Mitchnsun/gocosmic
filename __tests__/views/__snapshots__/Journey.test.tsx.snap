// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JourneyContent Component > should render the journey content with cosmic theme 1`] = `
<div>
  <div
    style="min-height: calc(100vh - var(--header-footer-height));"
  >
    <div
      class="fixed inset-0 w-full"
    >
      <div
        data-testid="three-canvas"
      >
        <div
          data-testid="moving-starfield"
        />
      </div>
    </div>
    <div
      class="px-4 py-48"
    >
      <h2
        class="text-ghost mb-4 text-center text-2xl font-bold lg:text-5xl"
      >
        Welcome to your Cosmic Journey
      </h2>
      <p
        class="flex items-center justify-center gap-2 text-gray-300 lg:text-xl"
      >
        You are now traveling through the stars
        <svg
          aria-hidden="true"
          class="text-ghost h-6 w-6 animate-bounce"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M9.315 7.584C12.195 3.883 16.695 1.5 21.75 1.5a.75.75 0 0 1 .75.75c0 5.056-2.383 9.555-6.084 12.436A6.75 6.75 0 0 1 9.75 22.5a.75.75 0 0 1-.75-.75v-4.131A15.838 15.838 0 0 1 6.382 15H2.25a.75.75 0 0 1-.75-.75 6.75 6.75 0 0 1 7.815-6.666ZM15 6.75a2.25 2.25 0 1 0 0 4.5 2.25 2.25 0 0 0 0-4.5Z"
            fill-rule="evenodd"
          />
          <path
            d="M5.26 17.242a.75.75 0 1 0-.897-1.203 5.243 5.243 0 0 0-2.05 5.022.75.75 0 0 0 .625.627 5.243 5.243 0 0 0 5.022-2.051.75.75 0 1 0-1.202-.897 3.744 3.744 0 0 1-3.008 1.51c0-1.23.592-2.323 1.51-3.008Z"
          />
        </svg>
      </p>
    </div>
  </div>
</div>
`;
